### PR TITLE
Run workflow

### LINKED ISSUE #1767: Refactor rust-gcc-diagnostics.cc
The functions defined within rust-gcc-diagnostics.cc should be moved into rust-diagnostics.cc

### PR DESCRIPTION
**Pull request by Parthib314**
Fixes #1767

### CODE REVIEW

File: run-clang-format.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
+#!/usr/bin/env python
```
Comment (CohenArthur) [inline run-clang-format.py:1]: I don't think this file should be part of this PR

**Review by CohenArthur — COMMENTED** [review]
The code changes look good to me.

Can you explain what `workflow_dispatch` does?
Also, could you rebase your commits into two commits? in the case of this PR, I'd like to see one commit addressing the CI changes and one addressing the refactor

File: run-clang-format.py (hunk 1 above)
Comment (Parthib314) [reply run-clang-format.py:1]: Oops, Sorry actually I was trying to run the clang-format test locally so I this file was created I forgot to remove it.

Comment (Parthib314) [reply run-clang-format.py:1]: Regarding `workflow_dispatch` :
By this, we can run the clang-format test manually along with other options. Actually, before submitting I was testing whether the clang-format test was running successfully or not. So I added this.
Screenshot from 2023-02-01 19-02-30
I was telling about the manual trigger `run workflow` button. Should I remove it??

Comment (CohenArthur) [reply run-clang-format.py]: As long as it still runs automatically on every PR I think it's a good change yeah :) it makes it easier for contributors to try and run clang-format if they can't install clang-format-10 on their machines. And we'll eventually upgrade clang-format 14 or 15

**Comment by powerboat9** [comment]
@Parthib314 you should probably rebase
@CohenArthur how do you satisfy the GNU Commit Format Checker?

**Comment by CohenArthur** [comment]
> @Parthib314 you should probably rebase @CohenArthur how do you satisfy the GNU Commit Format Checker?

You need to create a Changelog entry. You can generate that changelog entry with

```
git format-patch <commit_hash> --stdout -1 | ./contrib/mklog.py
```

we're working on adding automation to make this easier and more interactive :)

**Comment by CohenArthur** [comment]
But for now I'll take care of answering on pull-requests with the appropriate Changelog skeletons/templates. Which is why I'm waiting for Parthib to rebase this PR into two commits only :)

**Comment by Parthib314** [comment]
Should I make a fresh new PR??

**Comment by CohenArthur** [comment]
Hi @Parthib314 :) sorry for the delay in reviewing. Your first commit contains merge commits markers (`<<<<` and `>>>>`). Please merge the two commits together in one. If you're using git in the command line I can help you with the commands to do this.

Once this is done, update your branch (force-push on it) and ping me. I'll send you the Changelog you have to add to your commit.

**Review by CohenArthur — APPROVED** [review]
Thank you!

**Pull request merged**